source/module_hsolver/hsolver_lcao.cpp

```diff
@@ -165,89 +148,138 @@ void HSolverLCAO<T, Device>::solveTemplate(hamilt::Hamilt<T>* pHamilt,
         }
     }
 #ifdef __PEXSI
-    else if (this->method == "pexsi")
-    {
-        if (this->pdiagh != nullptr)
-        {
-            if (this->pdiagh->method != this->method)
-            {
+    else if (this->method == "pexsi") {
+        if (this->pdiagh != nullptr) {
+            if (this->pdiagh->method != this->method) {
                 delete[] this->pdiagh;
                 this->pdiagh = nullptr;
             }
             auto tem = dynamic_cast<DiagoPexsi<T>*>(this->pdiagh);
         }
-        if (this->pdiagh == nullptr)
-        {
+        if (this->pdiagh == nullptr) {
             DiagoPexsi<T>* tem = new DiagoPexsi<T>(this->ParaV);
             this->pdiagh = tem;
             // this->pdiagh = dynamic_cast<DiagoPexsi<T>*>(tem);
             this->pdiagh->method = this->method;
         }
     }
 #endif
-    else
-    {
-        ModuleBase::WARNING_QUIT("HSolverLCAO::solve", "This method of DiagH is not supported!");
+    else {
+        ModuleBase::WARNING_QUIT("HSolverLCAO::solve",
+                                 "This method of DiagH is not supported!");
     }
 
-    if (this->method == "cg_in_lcao")
-    {
+    if (this->method == "cg_in_lcao") {
         this->precondition_lcao.resize(psi.get_nbasis());
 
         using Real = typename GetTypeReal<T>::type;
         // set precondition
-        for (size_t i = 0; i < precondition_lcao.size(); i++)
-        {
+        for (size_t i = 0; i < precondition_lcao.size(); i++) {
             precondition_lcao[i] = 1.0;
         }
     }
 
-    /// Loop over k points for solve Hamiltonian to charge density
-    for (int ik = 0; ik < psi.get_nk(); ++ik)
-    {
-        /// update H(k) for each k point
-        pHamilt->updateHk(ik);
-
-        psi.fix_k(ik);
-
-        // solve eigenvector and eigenvalue for H(k)
-        this->hamiltSolvePsiK(pHamilt, psi, &(pes->ekb(ik, 0)));
+    if (Parallel_K2D<double>::get_instance().get_kpar() > 1) {
```

Comment:
Please don't do parallel operation in hsolver_lcao.cpp, you would better gather these code to another function to bracket your code.